python/private/pypi/whl_library.bzl

```diff
@@ -55,9 +55,31 @@ def _get_xcode_location_cflags(rctx):
         # This is a full xcode installation somewhere like /Applications/Xcode13.0.app/Contents/Developer
         # so we need to change the path to to the macos specific tools which are in a different relative
         # path than xcode installed command line tools.
-        xcode_root = "{}/Platforms/MacOSX.platform/Developer".format(xcode_root)
+        xcode_sdks_json = rctx.execute([
+            "xcrun",
+            "xcodebuild",
+            "-showsdks",
+            "-json",
+        ], environment = {
+            "DEVELOPER_DIR": xcode_root,
+        }).stdout
+        xcode_sdks = json.decode(xcode_sdks_json)
+        potential_sdks = [
+            sdk
+            for sdk in xcode_sdks
+            if "productName" in sdk and
+               sdk["productName"] == "macOS" and
+               "darwinos" not in sdk["canonicalName"]
+        ]
+
+        # Now we'll get two entries here (one for internal and another one for public)
+        # It shouldn't matter which one we pick.
+        xcode_sdk_path = potential_sdks[0]["sdkPath"]
+    else:
+        xcode_sdk_path = "{}/SDKs/MacOSX.sdk".format(xcode_root)
```

Comment:
Thinking out loud, is this path absolute? I am wondering if we should have the path in the `rules_python_internal` where we then can include the path by importing it as a constant.